packages/autorest.python/autorest/codegen/serializers/model_serializer.py

```diff
@@ -72,16 +89,57 @@ def get_properties_to_initialize(model: ModelType) -> List[Property]:
             properties_to_initialize = model.properties
         return properties_to_initialize
 
+    @staticmethod
+    def get_properties_to_declare(model: ModelType) -> List[Property]:
+        if model.parents:
+            properties_to_declare = list(
+                {
+                    p.client_name: p
+                    for bm in model.parents
+                    for p in model.properties
+                    if p not in cast(ModelType, bm).properties
+                }.values()
+            )
+        else:
+            properties_to_declare = model.properties
+        return properties_to_declare
+
     def declare_model(self, model: ModelType) -> str:
         basename = (
             "msrest.serialization.Model"
             if self.code_model.options["client_side_validation"]
+            else "_model_base.Model"
+            if self.code_model.options["models_mode"] == "dpg"
             else "_serialization.Model"
         )
         if model.parents:
             basename = ", ".join([cast(ModelType, m).name for m in model.parents])
+        if (
+            self.code_model.options["models_mode"] == "dpg"
+            and model.discriminator_value
+        ):
+            basename += f", discriminator='{model.discriminator_value}'"
         return f"class {model.name}({basename}):{model.pylint_disable}"
 
+    @staticmethod
+    def declare_property(prop: Property) -> List[str]:
+        attribute_key = prop.rest_api_name.replace(".", "\\\\.")
```

Comment:
Right now the code isn't very cleaan, because we're doing this attribute key replacing both in `property.py` and in `declare_property`. It sems like msrest is using the `attribute_map` in `property.py`, but cadl is using this separate function in `model_serializer.py`. We want to model type and property python class definitions to be completely agnostic over whether we are serializing a cadl model or a msrest model. The cadl / msrest specific serialization logic should be done in the `model_serializer` and it should be very clear what code is msrest specific and what code is cadl specific

Reply:
Yes, I see the `attribute_map` in `property.py` is actually for serialization. So I moved it to `model_serializer.py` as `declare_msrest_attribute_map`.